### PR TITLE
Drop tech preview v1alpha1 promtail collector API

### PR DESCRIPTION
**Pull request by periklis**
### Description
This PR removes the v1alpha1 collector API for release 4.6 to eliminate all remaining tech preview features in cluster-logging operator.

Ref: https://issues.redhat.com/browse/LOG-841

/cc @jcantrill @alanconway

### CODE REVIEW

**Comment by jcantrill** [comment]
WFM as long as @brancz team is not making use of it

**Comment by brancz** [comment]
We are not. We just used it for some experiments in the past.

File: olm_deploy/scripts/catalog-deploy.sh [hunk 1]
```diff
@@ -29,13 +27,12 @@ echo "cluster logging operator: ${IMAGE_CLUSTER_LOGGING_OPERATOR}"
...
 echo "curator5: ${IMAGE_LOGGING_CURATOR5}"
 echo "fluentd: ${IMAGE_LOGGING_FLUENTD}"
-echo "promtail: ${IMAGE_PROMTAIL}"
 echo "elastic6: ${IMAGE_ELASTICSEARCH6}"
 echo "kibana: ${IMAGE_LOGGING_KIBANA6}"
 
 echo "In namespace: ${CLUSTER_LOGGING_OPERATOR_NAMESPACE}"
 
-if oc get project ${CLUSTER_LOGGING_OPERATOR_NAMESPACE} > /dev/null 2>&1 ; then
+if oc get prjeoct ${CLUSTER_LOGGING_OPERATOR_NAMESPACE} > /dev/null 2>&1 ; then
```
Comment (ewolinetz) [inline olm_deploy/scripts/catalog-deploy.sh:35]: spelling

**Pull request merged**